### PR TITLE
Delete message

### PR DESCRIPTION
**Pull request by davidsoderberg**
- **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
An endpoint to delete messages
- **What is the current behavior?** (You can also link to an open issue here)
The endpoint does not exist today
- **What is the new behavior (if this is a feature change)?**
An endpoint to delete messages
- **Other information**:
This can be nice to have so our users can delete test messages inside of the notification center specifically

### CODE REVIEW

File: libs/dal/src/repositories/message/message.repository.ts [hunk 1]
```diff
@@ -220,4 +224,18 @@ export class MessageRepository extends BaseRepository<MessageEntity> {
       data: this.mapEntities(response),
     };
   }
+
+  async delete(query: FilterQuery<MessageEntity & Document>) {
+    const message = await this.findOne({ _id: query._id });
+    if (!message) {
+      throw new DalException(`Could not find feed with id ${query._id}`);
```
Comment (djabarovgeorge) [inline libs/dal/src/repositories/message/message.repository.ts:231]: Small suggestion I think there is a typo
```suggestion
      throw new DalException(`Could not find a message with id ${query._id}`);
```

**Review by djabarovgeorge — COMMENTED** [review]
Loved the new endpoint we could add this option to the notification-center in the future 🤩

File: apps/api/src/app/messages/messages.controller.ts [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+  @ExternalApiAccessible()
+  @UseGuards(JwtAuthGuard)
+  @ApiOkResponse({
+    type: DeleteMessageResponseDto,
+  })
+  @ApiOperation({
+    summary: 'Delete message',
+    description: 'Deletes a message entity from the Novu platform',
+  })
+  async removeSubscriber(
```
Comment (LetItRock) [inline apps/api/src/app/messages/messages.controller.ts:25]: Looks like you forgot to change the name here ;)

**Review by scopsy — APPROVED** [review]
Looks great other than what @LetItRock mentioned 🚀

**Pull request merged**